packages/webapp/src/delegates/components/my-delegation.tsx

```diff
@@ -0,0 +1,174 @@
+                                    <DelegateChip
+                                        member={members.find(
+                                            (d) =>
+                                                d.account === delegate.account
+                                        )}
+                                        level={index + 1}
+                                    />
+                                </div>
+                            </>
+                        )}
+                        <MyDelegationArrow />
+                    </div>
+                ))}
+            {[...Array(numLevelsWithNoRepresentation)].map((v, idx) => (
+                <div className="-mt-px" key={idx}>
+                    {/* TODO: We'll need level headings for these too */}
+                    <DelegateChip />
+                    <MyDelegationArrow />
+                </div>
+            ))}
+        </>
+    );
+};
+
+const ChiefDelegates = ({
+    electionState,
+}: {
+    electionState?: ElectionState;
+}) => {
+    const {
+        data: membersStats,
+        isLoading: isLoadingMemberStats,
+        isError: isErrorMemberStats,
+    } = useMemberStats();
+
+    const allChiefAccountNames = electionState?.board || [];
+    // Get EdenMember data, unwrap the QueryResults[] into an EdenMember[], and filter out non-existent members
+    const chiefsAsMembers = useMemberListByAccountNames(allChiefAccountNames)
+        .map((chiefQR) => chiefQR.data)
+        .filter((el) => Boolean(el));
+
+    const nftTemplateIds = chiefsAsMembers.map(
+        (member) => member!.nft_template_id
+    );
+
+    const {
+        data: memberData,
+        isLoading: isLoadingMemberData,
+        isError: isErrorMemberData,
+    } = useQuery({
+        ...queryMembers(1, allChiefAccountNames.length, nftTemplateIds),
+        staleTime: Infinity,
+        enabled: Boolean(chiefsAsMembers?.length),
+    });
+
+    const isLoading = isLoadingMemberStats || isLoadingMemberData;
+    const isError = isErrorMemberStats || isErrorMemberData;
+
+    if (isLoading) {
+        return <LoadingContainer />;
+    }
+
+    if (isError || !electionState || !memberData || !membersStats) {
+        return <ErrorLoadingDelegation />;
+    }
+
+    // TODO: Handle the no-election-has-ever-happened scenario (just after genesis induction is complete)
+    const headChiefAsEdenMember = chiefsAsMembers.find(
+        (d) => d?.account === electionState.lead_representative
+    );
+    const headChiefAsMemberData = memberData.find(
+        (d) => d?.account === electionState.lead_representative
+    );
+
+    if (!headChiefAsEdenMember || !headChiefAsMemberData) {
+        return <ErrorLoadingDelegation />;
+    }
+
+    return (
+        <>
+            <div>
+                <LevelHeading className="border-b">
+                    Chief Delegates
+                </LevelHeading>
+                <MembersGrid members={memberData}>
```

Comment:
Ok just to clarify: Chief Delegates will show all delegates including the Head Chief. and
Will the logged-in user's rep have a delegate badge in the UI so the user can see who their delegate is?

Reply:
For now, all Chief Delegates have a delegate badge. So there's room for improvement here. There shouldn't be any changes to logic on this page in this PR, though. I'm aiming to maintain the status quo.

Reply:
It looks like this is a pretty strict refactor / adding of very limited items like Loading and Error states, so... I'll just consider my comments on this comments-to-self. :)